lib/Frontend/CompilerInvocation.cpp

```diff
@@ -908,7 +908,16 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
   
   for (const Arg *A : make_range(Args.filtered_begin(OPT_D),
                                  Args.filtered_end())) {
-    Opts.addCustomConditionalCompilationFlag(A->getValue());
+    StringRef name = A->getValue();
+    if (Lexer::isIdentifier(name))
+      Opts.addCustomConditionalCompilationFlag(name);
+    else if (name.find('=') != StringRef::npos)
+      Diags.diagnose(SourceLoc(),
```

Comment:
That's not a supported configuration anyway. Anyone using -frontend is supposed to be a compiler developer already. I suppose we could assert it, though, just to have a weak catch.